### PR TITLE
cgen: fix array init with fields (fix #16596)

### LINKED ISSUE #16596: problems with array allocation.
### Describe the bug

when trying to edit element of an array v panics and stops execution.
this happens when only the type of array is given along with the length.
when trying to edit element (probably increase in size) v panics.

### Expected Behavior

be able to edit element with no errors.

### Current Behavior

```
V panic: malloc(0 <= 0)
v hash: 7e9e2ff
/tmp/v_1000/l.2758315796347022139.tmp.c:6842: at _v_panic: Backtrace
/tmp/v_1000/l.2758315796347022139.tmp.c:7048: by _v_malloc
/tmp/v_1000/l.2758315796347022139.tmp.c:5956: by array_ensure_cap
/tmp/v_1000/l.2758315796347022139.tmp.c:6304: by array_push
/tmp/v_1000/l.2758315796347022139.tmp.c:12647: by datatypes__Stack_T_string_push_T_string
/tmp/v_1000/l.2758315796347022139.tmp.c:12755: by main__Fluid_T_string_to_stack_T_string
/tmp/v_1000/l.2758315796347022139.tmp.c:12769: by main__main
/tmp/v_1000/l.2758315796347022139.tmp.c:13144: by main
```

### Reproduction Steps

```v
import datatypes

struct Fluid[T] {
	mut:
		elements datatypes.Queue[T]
		switch bool // 0 by default
		selements datatypes.Stack[T]
}
fn (mut fluid Fluid[T]) push(item T) {
	if fluid.switch {
		fluid.selements.push(item)
	} else {
		fluid.elements.push(item)
	}
}

fn (mut fluid Fluid[T]) pop() !T {
	if fluid.switch {
		return fluid.selements.pop()!
	} else{
		return fluid.elements.pop()!
	}
}

fn (fluid Fluid[T]) str() string {
	if fluid.switch {
		return fluid.selements.str()
	} else{
		return fluid.elements.str()
	}
}

fn (fluid Fluid[T]) peek() !T {
	if fluid.switch {
		return fluid.selements.peek()!
	} else{
		return fluid.elements.peek()!
	}
}

// this function converts queue to stack
fn (mut fluid Fluid[T]) to_stack() !bool {
	if !fluid.switch {
		// convert queue to stack
		for i in 1..fluid.elements.len() + 1 {
			fluid.selements.push(fluid.elements.index(fluid.elements.len() - i)!)
		}
		fluid.switch = true
	}
	return true
}

fn main() {
	
	mut arr := []Fluid<string>{len: 10}
	for i in 0..arr.len {
		// add str to queue
		arr[i].push("${10+i}")
	}		
	arr[1].to_stack()!
}```

### Possible Solution

i fixed the issue by manually defining array with all it's elements.
```v
mut arr := [Fluid[string]{},Fluid[string]{}, Fluid[string]{}, Fluid[string]{},Fluid[string]{}, Fluid[string]{}, Fluid[string]{},Fluid[string]{}, Fluid[string]{}]

```

### Additional Information/Context

_No response_

### V version

V 0.3.2 7e9e2ff

### Environment details (OS name and version, etc.)

``` OS: linux, Pop!_OS 22.04 LTS Processor: 4 cpus, 64bit, little endian, Intel(R) Core(TM) i5-4210U CPU @ 1.70GHz CC version: cc (Ubuntu 11.3.0-1ubuntu1~22.04) 11.3.0  getwd: /home/mp/AOC/day5 vmodules: /home/mp/.vmodules vroot: /home/mp/Downloads/v vexe: /home/mp/Downloads/v/v vexe mtime: 2022-12-05 13:24:51 is vroot writable: true is vmodules writable: true V full version: V 0.3.2 7e9e2ff  Git version: git version 2.34.1 Git vroot status: weekly.2022.48-29-g7e9e2ff4 (1 commit(s) behind V master) .git/config present: true thirdparty/tcc status: thirdparty-linux-amd64 12f392c3 ```

### PR DESCRIPTION
**Pull request by yuyi98**
This PR fix array init with fields (fix #16596).

- Fix array init with fields.
- Add test.

```v
import datatypes

struct Fluid[T] {
mut:
	elements  datatypes.Queue[T]
	switch    bool // 0 by default
	selements datatypes.Stack[T]
}

fn (mut fluid Fluid[T]) push(item T) {
	if fluid.switch {
		fluid.selements.push(item)
	} else {
		fluid.elements.push(item)
	}
}

fn (mut fluid Fluid[T]) pop() !T {
	if fluid.switch {
		return fluid.selements.pop()!
	} else {
		return fluid.elements.pop()!
	}
}

fn (fluid Fluid[T]) str() string {
	if fluid.switch {
		return fluid.selements.str()
	} else {
		return fluid.elements.str()
	}
}

fn (fluid Fluid[T]) peek() !T {
	if fluid.switch {
		return fluid.selements.peek()!
	} else {
		return fluid.elements.peek()!
	}
}

// this function converts queue to stack
fn (mut fluid Fluid[T]) to_stack() !bool {
	if !fluid.switch {
		// convert queue to stack
		for i in 1 .. fluid.elements.len() + 1 {
			fluid.selements.push(fluid.elements.index(fluid.elements.len() - i)!)
		}
		fluid.switch = true
	}
	return true
}

fn main() {
	mut arr := []Fluid[string]{len: 10}
	for i in 0 .. arr.len {
		// add str to queue
		arr[i].push('${10 + i}')
	}
	println(arr)
	ret := arr[1].to_stack()!
	assert ret
}

PS D:\Test\v\tt1> v run .
[['10'], ['11'], ['12'], ['13'], ['14'], ['15'], ['16'], ['17'], ['18'], ['19']]
```

### CODE REVIEW

File: vlib/regex/regex.v [hunk 1]
```diff
@@ -460,7 +460,7 @@ const (
 
 struct CharClass {
 mut:
-	cc_type   int = regex.cc_null // type of cc token
+	cc_type   int         // type of cc token
```
Comment (spytheman) [inline vlib/regex/regex.v:463]: hm, I do not see how is that related?

Comment (spytheman) [reply vlib/regex/regex.v:463]: oh, it seems to be just a cleanup; regex.cc_null is 0

Comment (yuyi98) [reply vlib/regex/regex.v:463]: Currently type_default() has some problems with it.
